src/proposer.ts

```diff
@@ -121,6 +130,246 @@ let s: ReturnType<typeof strings>
 // Initialization flag
 let isInitialized = false
 
+// ~7 days at ~12s blocks
+const APPROX_7D_BLOCKS = 50400
+/**
+ * Validates that a vault ID exists on-chain by checking it is within range
+ * [1, nextVaultId - 1]. Throws if invalid or out of range.
+ */
+export async function validateVaultIdOnChain(vaultId: number): Promise<void> {
+	// Basic sanity
+	if (!Number.isInteger(vaultId) || vaultId < 1) {
```

Comment:
We should allow `0` as the `vaultId` since that's the protocol's vault but disallow any `< 0`.